### PR TITLE
add focus to header on nav

### PR DESCRIPTION
**Pull request by dcmcand**
**Description of change**
added focus to header on nav.

**How to test**
Verify that when moving through pages of a report, focus is set to the header.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/212

**Checklist**

- [X] Meets issue criteria
- [X] Code tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Review by jasalisbury — COMMENTED** [review]
I just addressed this same issue because of accessibility concerns, but this is a way better solution then what I came up with. I think this solution can be a _little_ better if we do the following:

Add an empty dependecy array to the `useEffect` so the focus only happens the first time the header is rendered:
```
  useEffect(() => {
    document.getElementsByClassName('usa-step-indicator__heading-text')[0].focus();
  }, []);
```
Add a `key` to the `<NavigationHeader>` component in `frontend/src/components/Navigator/index.js` ~line 175 so the header is re-renedered every time the label changes:

```
<NavigatorHeader
  key={page.label}
  label={page.label}
/>
```

These two changes prevent focus changing to the header moments before the new form page is loaded.

File: frontend/src/components/Navigator/components/NavigatorHeader.js [hunk 1]
```diff
@@ -5,24 +5,25 @@
...
-  });
+  }, []);
 
   return (
     <div className="usa-step-indicator__header">
       {/* <style>{`:focus{ outline: solid 1px red}`}</style> */}
       <h2 className="usa-step-indicator__heading">
         {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
-        <span tabIndex="0" className="usa-step-indicator__heading-text">{label}</span>
+        <span tabIndex="0" key={key} className="usa-step-indicator__heading-text">{label}</span>
```
Comment (jasalisbury) [inline frontend/src/components/Navigator/components/NavigatorHeader.js:18]: You should be able to remove `key={key}`

File: frontend/src/components/Navigator/components/NavigatorHeader.js [hunk 2]
```diff
@@ -5,24 +5,25 @@
...
       </h2>
     </div>
   );
 };
 
 NavigatorHeader.propTypes = {
   label: PropTypes.string.isRequired,
+  key: PropTypes.string.isRequired,
```
Comment (jasalisbury) [inline frontend/src/components/Navigator/components/NavigatorHeader.js:26]: Can also remove this line

Comment (dcmcand) [reply frontend/src/components/Navigator/components/NavigatorHeader.js]: And thus, my react ignorance is revealed :smile:

**Review by jasalisbury — APPROVED** [review]
:rocket:

File: frontend/src/components/Navigator/components/NavigatorHeader.js (hunk 2 above)
Comment (jasalisbury) [reply frontend/src/components/Navigator/components/NavigatorHeader.js]: Keys are a special prop that tells gives the component an id, when that id changes react knows it needs to re-render the component. Usually they are used in lists so react can track which items need to be updated/removed when a list changes https://reactjs.org/docs/lists-and-keys.html#keys

Comment (dcmcand) [reply frontend/src/components/Navigator/components/NavigatorHeader.js]: Cool, thanks for the explanation.

File: frontend/src/components/Navigator/components/NavigatorHeader.js [hunk 3]
```diff
@@ -2,16 +2,24 @@
...
-  </div>
-);
+const NavigatorHeader = ({ label }) => {
+  useEffect(() => {
+    document.getElementsByClassName('usa-step-indicator__heading-text')[0].focus();
+  }, []);
+
+  return (
+    <div className="usa-step-indicator__header">
+      {/* <style>{`:focus{ outline: solid 1px red}`}</style> */}
```
Comment (rahearn) [inline frontend/src/components/Navigator/components/NavigatorHeader.js:15]: Is this needed anymore? Might as well clean it up.

**Review by rahearn — APPROVED** [review]
Works real well for me! :shipit:

**Pull request merged**